### PR TITLE
Fixing web view authentication

### PR DESCRIPTION
**Pull request by jleandroperez**
#### Steps:
1. Fresh install WPiOS
2. Launch the app, and log into your account
3. Tap over the Notifications tab
4. Find a Stats-Y notification for a *private* site and open it
5. Tap over the site name

As a result, WebViewController is expected to be onscreen, and the blog should be visible. No extra authentication should be required.

Needs Review: @diegoreymendez  ||  @sendhil (Thanks in advance!)

### CODE REVIEW

File: WordPress/Classes/Utility/WPURLRequest.m [hunk 1]
```diff
@@ -21,14 +21,19 @@ + (NSURLRequest *)requestForAuthenticationWithURL:(NSURL *)loginUrl
...
+    NSString *hostname = loginUrl.host;
+    
+    // Let's make sure we don't send OAuth2 tokens outside of wordpress.com
     if (![hostname isEqualToString:@"wordpress.com"] && ![hostname hasSuffix:@".wordpress.com"]) {
-        // Let's make sure we don't send OAuth2 tokens outside of wordpress.com
         bearerToken = nil;
+        
+    // Enforce HTTPS for WordPress.com Sites
+    } else if ([loginUrl.scheme isEqual:@"http"]) {
+        loginUrl = [[NSURL alloc] initWithScheme:@"https" host:loginUrl.host path:loginUrl.path];
```
Comment (diegoreymendez) [inline WordPress/Classes/Utility/WPURLRequest.m]: Careful with this... this does not maintain the query field of a URL.  I've tested with the following code and the resulting URL is not the same:

```objective-c
    NSURL *url = [NSURL URLWithString:@"http://diego.com/testing/path?no=joking&is=true"];
    NSURL *url2 = [[NSURL alloc] initWithScheme:url.scheme host:url.host path:url.path];
```

**Comment by diegoreymendez** [comment]
@jleandroperez - Just one comment on the code.

@sendhil - Jorge mentioned you have Stats-Y notifications for a private site, could you test these changes?  I'm not sure about how to get such a notification myself.

**Comment by sendhil** [comment]
@diegoreymendez - sure, will do now.

**Comment by sendhil** [comment]
@jleandroperez works for me. Could you address @diegoreymendez's comment?

**Comment by jleandroperez** [comment]
@diegoreymendez + @sendhil: Thanks for the review gentleman!. Ready for a second round, http parameters should now be carried over.

**Comment by diegoreymendez** [comment]
@jleandroperez - On the code side of things I think it's looking good now. :shipit: on my end... but make sure to check with @sendhil too since I couldn't test this feature.

**Comment by sendhil** [comment]
Works for me.

**Comment by jleandroperez** [comment]
Thank you @diegoreymendez  ++ @sendhil :smile: !!

**Pull request merged**